core/src/main/scala/org/apache/spark/util/random/SamplingUtils.scala

```diff
@@ -52,16 +52,20 @@ private[spark] object SamplingUtils {
       (trimReservoir, i)
     } else {
       // If input size > k, continue the sampling process.
+      var l = i.toLong
       val rand = new XORShiftRandom(seed)
       while (input.hasNext) {
         val item = input.next()
-        val replacementIndex = rand.nextInt(i)
+        val replacementIndex = l < Int.MaxValue match {
```

Comment:
I think the change is fine except that the `case true` business strikes me as pointlessly verbose Scala. `if else` is much clearer.